lib/elixir/lib/file.ex

```diff
@@ -931,6 +931,11 @@ defmodule File do
   option when opening the file and then all other functions from
   `IO` are available, since they work directly with Unicode data.
 
+  `modes_or_function` can either be a list of modes or a function. If it's a
+  list, it's considered to be a list of modes (that are documented below). If
+  it's a function, then it's equivalent to calling `open(path, [],
+  modes_or_function)`. See the documentation for `open/3` for more information.
```

Comment:
Sorry, I misexplained there, I meant that how the code is written affected how I thought about the docs and wrote the docs 😄 I will rephrase, better to be ultra-clear.